testing/sharedtesthelpers/helpers.go

```diff
@@ -0,0 +1,236 @@
+	// and will be translated to an absolute path
+	FakeScanRoot string
+	FakeFileInfo *fakefs.FakeFileInfo
+}
+
+type ScanInputWrapper struct {
+	fileHandle *os.File
+	ScanInput  filesystem.ScanInput
+}
+
+func (siw ScanInputWrapper) Close() {
+	siw.fileHandle.Close()
+}
+
+// Generate FileInfoMock will either use the fake file information if fakeFileInfo is true,
+// otherwise try to run os.Stat on the path passed in and fail if the file does not exist
+func GenerateFileInfoMock(t *testing.T, config ScanInputMockConfig) fs.FileInfo {
+	t.Helper()
+
+	if config.FakeFileInfo != nil {
+		ret := *config.FakeFileInfo
+		ret.FileName = filepath.Base(config.Path)
+
+		return ret
+	}
+
+	fileInfo, err := os.Stat(config.Path)
+	// It is intended that sometimes the config points to a path that does not exist
+	// fileInfo will be nil in those cases
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		t.Fatalf("Can't stat test fixture '%s' because '%s'", config.Path, err)
+	}
+
+	return fileInfo
+}
+
+// GenerateScanInputMock will try to open the file locally, and fail if the file doesn't exist
+func GenerateScanInputMock(t *testing.T, config ScanInputMockConfig) ScanInputWrapper {
+	t.Helper()
+
+	var scanRoot string
+	if filepath.IsAbs(config.FakeScanRoot) {
+		scanRoot = config.FakeScanRoot
+	} else {
+		workingDir, err := os.Getwd()
+		if err != nil {
+			t.Fatalf("Can't get working directory because '%s'", workingDir)
+		}
+		scanRoot = filepath.Join(workingDir, config.FakeScanRoot)
+	}
+
+	f, err := os.Open(filepath.Join(scanRoot, config.Path))
+	if err != nil {
+		t.Fatalf("Can't open test fixture '%s' because '%s'", config.Path, err)
+	}
+	info, err := f.Stat()
+	if err != nil {
+		t.Fatalf("Can't stat test fixture '%s' because '%s'", config.Path, err)
+	}
+
+	return ScanInputWrapper{
+		fileHandle: f,
+		ScanInput: filesystem.ScanInput{
+			FS:     os.DirFS(scanRoot).(scalibrfs.FS),
+			Path:   config.Path,
+			Root:   scanRoot,
+			Reader: f,
+			Info:   info,
+		},
+	}
+}
+
+// Form returns the singular or plural form that should be used based on the given count
+func Form(count int, singular, plural string) string {
+	if count == 1 {
+		return singular
+	}
+
+	return plural
+}
+
+type TestTableEntry struct {
+	Name              string
+	InputConfig       ScanInputMockConfig
+	WantInventory     []*extractor.Inventory
+	WantErrIs         error
+	WantErrContaining string
+}
+
+// ExtractionTester tests common properties of a extractor, and returns the raw values from running extract
+func ExtractionTester(t *testing.T, extractor filesystem.Extractor, tt TestTableEntry) ([]*extractor.Inventory, error) {
+	t.Helper()
+
+	wrapper := GenerateScanInputMock(t, tt.InputConfig)
+	got, err := extractor.Extract(context.Background(), &wrapper.ScanInput)
+	wrapper.Close()
+	if tt.WantErrIs != nil {
+		ExpectErrIs(t, err, tt.WantErrIs)
+	}
+	if tt.WantErrContaining != "" {
+		ExpectErrContaining(t, err, tt.WantErrContaining)
```

Comment:
In SCALIBR we've been using  cmp.Equal / cmpopts.EquateErrors to compare errors. I wonder if this could be ported to a struct similar to AnyError so we can just do a normal cmp instead of needing a custom function - https://github.com/google/go-cmp/blob/v0.6.0/cmp/cmpopts/equate.go#L128

Might be out of scope for this CL but something we may want to track somewhere

Reply:
Had a quick look into this, we probably can do something similar, but because we only use it in one place, I just inlined the check here, and removed the assertion. 

I think the current version is now pretty clear, can you take another look?

Reply:
As discussed offline I think it'd still make sense to use cmpopts.EquateErrors for this with custom comparison structs - that'd allow us to remove EquateErrors and we'd only need one ExpectedError field in the test table lib.

Reply:
Done, added a new error type ContainErrStr which will match with errors containing a specific string.